### DISCUSSION TITLE
Formatting of long lines with explicit and implicit parameters

### DISCUSSION THREAD

**Discussion by jozic** [scalameta/scalafmt#3722] [Q&A]
Hello All,

I would like my methods/constructors with several sets of parameters (explicit and implicit) when a line limit is reached to look like
```scala
def method(
  param1: String,
  param2: Int,
  param3: Double
)(implicit param4: Boolean): Unit = 
```
by default scalafmt breaks the line on first possible symbol, like
```scala
def method(param1: String, param2: Int, param3: Double)(
  implicit param 3: Boolean
): Unit = 
```

and i can't find a configuration to make it my way.
Does anyone know how to achieve this?
Thanks a lot!

**Comment by kitbellew** [comment]
have you had a chance to read through https://scalameta.org/scalafmt/docs/configuration.html?

**Reply by jozic** [reply]
I did, several times in the past, but never was successful in achieving desired formatting. That's why I've asked here.

I know docs are updated frequently, so I will try again soon, but if someone already knows how to get what I need, that would be handy.

**Comment by kitbellew** [comment]
newlines.source, perhaps?
